src/cascade/input_data/db/bundle.py

```diff
@@ -0,0 +1,327 @@
+import logging
+
+import pandas as pd
+
+from cascade.core.db import cursor, connection
+
+CODELOG = logging.getLogger(__name__)
+
+# FIXME: There is a shared function that get's the official mapping, I think. Or an sql query at least.
+MEASURES = {6: "incidence", 9: "mtexcess"}
+
+
+def _bundle_is_frozen(execution_context):
+    """Checks if data for the current model_version_id already exists in tier 3.
+    """
+
+    model_version_id = execution_context.parameters.model_version_id
+    query = f"""
+    select exists(
+             select * from epi.t3_model_version_dismod
+             where model_version_id = {model_version_id}
```

Comment:
Is one of these methods preferred?  
i.e. where model_version_id = {model_version_id}   vs.  where model_version_id = %(model_version_id)s

Reply:
Yes. This is a mistake. It should be %(model_version_id)s. The difference is in when the substitution happens. With the f-string version the substitution is done by python without any awareness of SQL. In the other version the substitution is done by the SQL client itself which allows it to protect itself from values which might have bad escaping in them or other things that could unexpectedly render the SQL statement incorrect.

In applications which are more exposed to untrusted users than us, this is a serious concern because the f-string version might allow someone to inject malicious code into the query. In our case, the worst that is likely is that someone will accidentally stick garbage in there and it'll fail to parse but not cause damage.

Rarely it is still necessary to use f-string formatting when dynamically setting table or column names rather than values, but that's uncommon. I think there's one example somewhere in this code.